taskcluster_proxy_test.go

```diff
@@ -7,16 +7,23 @@ import (
 )
 
 func TestTaskclusterProxy(t *testing.T) {
+	if os.Getenv("TASKCLUSTER_CLIENT_ID") == "" ||
+		os.Getenv("TASKCLUSTER_ACCESS_TOKEN") == "" ||
+		os.Getenv("TASKCLUSTER_ROOT_URL") == "" {
+		t.Skip("Skipping test since TASKCLUSTER_{CLIENT_ID,ACCESS_TOKEN,ROOT_URL} env vars not set")
+	}
+
 	defer setup(t)()
 	payload := GenericWorkerPayload{
 		Command: append(
 			append(
 				goEnv(),
+				// long enough to reclaim and get new credentials
 				sleep(12)...,
 			),
 			goRun(
 				"curlget.go",
-				fmt.Sprintf("http://localhost:%v/queue/v1/task/KTBKfEgxR5GdfIIREQIvFQ/runs/0/artifacts/SampleArtifacts/_/X.txt", config.TaskclusterProxyPort),
+				fmt.Sprintf("$TASKCLUSTER_PROXY_URL/queue/v1/task/KTBKfEgxR5GdfIIREQIvFQ/runs/0/artifacts/SampleArtifacts/_/X.txt"),
```

Comment:
This warrants a comment that this is literally replaced by `curlget.go` program, otherwise it is confusing, since there is no shell involved.

I would have preferred a more general solution like using `${TASKCLUSTER_PROXY_URL}` and then in `curlget.go` expanding all occurrences of `${...}` using the process environment.

Reply:
I'll add the comments. `curlget.go` is only used by this test, so I don't think it warrants generalization.